android/src/main/java/com/rnim/rn/audio/AudioRecorderManager.java

```diff
@@ -201,43 +229,80 @@ public void stopRecording(Promise promise){
-    // Added this function to have the same api for android and iOS, stops recording now
-    stopRecording(promise);
+  public void pauseRecording(Promise promise) {
+    if (!isPauseResumeCapable || pauseMethod==null) {
+      logAndRejectPromise(promise, "RUNTIME_EXCEPTION", "Method not available on this version of Android.");
+      return;
+    }
+
+    if (!isPaused) {
+      try {
+        pauseMethod.invoke(recorder);
+        stopWatch.stop();
+      } catch (InvocationTargetException | RuntimeException | IllegalAccessException e) {
+        e.printStackTrace();
+        logAndRejectPromise(promise, "RUNTIME_EXCEPTION", "Method not available on this version of Android.");
+        return;
+      }
+    }
+
+    isPaused = true;
+    promise.resolve(null);
+  }
+
+  @ReactMethod
+  public void resumeRecording(Promise promise) {
+    if (!isPauseResumeCapable || resumeMethod == null) {
+      logAndRejectPromise(promise, "RUNTIME_EXCEPTION", "Method not available on this version of Android.");
+      return;
+    }
+
+    if (isPaused) {
+      try {
+        resumeMethod.invoke(recorder);
+        stopWatch.start();
+      } catch (InvocationTargetException | RuntimeException | IllegalAccessException e) {
+        e.printStackTrace();
+        logAndRejectPromise(promise, "RUNTIME_EXCEPTION", "Method not available on this version of Android.");
+        return;
+      }
+    }
+    
+    isPaused = false;
+    promise.resolve(null);
   }
 
   private void startTimer(){
-    stopTimer();
     timer = new Timer();
     timer.scheduleAtFixedRate(new TimerTask() {
       @Override
       public void run() {
-        WritableMap body = Arguments.createMap();
-        body.putInt("currentTime", recorderSecondsElapsed);
-        if(meteringEnabled){
-            int amplitude = recorder.getMaxAmplitude();
-                if (amplitude == 0) {
-                    body.putInt("currentMetering", -160);//The first call - absolutely silence  
-                } else {
-                    //db = 20 * log10(peaks/ 32767); where 32767 - max value of amplitude in Android, peaks - current value
-                    body.putInt("currentMetering", (int) (20 * Math.log(((double) amplitude) / 32767d)));
-                }
+        if (!isPaused) {
+          WritableMap body = Arguments.createMap();
+          body.putDouble("currentTime", stopWatch.getTimeSeconds());
+          if(meteringEnabled){
+              int amplitude = recorder.getMaxAmplitude();
+                  if (amplitude == 0) {
+                      body.putDouble("currentMetering", -160);//The first call - absolutely silence
+                  } else {
+                      //db = 20 * log10(peaks/ 32767); where 32767 - max value of amplitude in Android, peaks - current value
+                      body.putDouble("currentMetering", (double) (20 * Math.log(((double) amplitude) / 32767d)));
+                  }
+          }
+          sendEvent("recordingProgress", body);
```

Comment:
Check this out if this makes sense since this is where i had some merge conflicts @FireyFly @markusekblad